requre/online_replacing.py

```diff
@@ -15,132 +15,153 @@
     TEST_METHOD_REGEXP,
 )
 from requre.helpers.requests_response import RequestResponseHandling
-from requre.utils import get_datafile_filename
+from requre.utils import get_datafile_filename, get_module_of_previous_context
 from requre.helpers.guess_object import Guess
 
 logger = logging.getLogger(__name__)
 
 
+class ModuleRecord:
+    def __init__(self, what, parent, original, replacement):
+        self.what = what
+        self.parent = parent
+        self.original = original
+        self.replacement = replacement
+        self.caller_module = get_module_of_previous_context()
+
+
+def _match(
+    what, module, cassette, decorate, replace, module_record_list: List[ModuleRecord]
```

Comment:
Do you think it will be possible to decompose this function? It's too long to be nicely readable....

Reply:
I understand, but this is also decomposing of original bigger one, and I can decompose it more later, but this is nothing what I now need for my next work. I'll create issue for this